client/js/components/SelfGuidedItem/SelfGuidedItem.js

```diff
@@ -6,56 +6,81 @@ import {
-      source={{uri: selfGuidedItem.image}}
-      style={styles.backgroundImage}>
-      <View style={styles.infoContainer}>
-        <Text style={styles.title}>{selfGuidedItem.title}</Text>
-        <Text style={styles.location}>{selfGuidedItem.location}</Text>
-        <View style={styles.flexRow}>
-          <View style={styles.flexRow}>
-            <View style={styles.flexColumn}>
-              <Text style={styles.subTitles}>Difficulty</Text>
-              <Text style={styles.infobox}>{selfGuidedItem.difficulty}</Text>
-            </View>
-            <View style={styles.flexColumn}>
-              <Text style={styles.subTitles}>Distance</Text>
-              <Text style={styles.infobox}>{selfGuidedItem.distance}</Text>
+  <SaveEventContext.Consumer>
+    {({savedIds, addSaveId, removeSaveId}) => (
+      <TouchableOpacity
+        onPress={() => {
+          if (selfGuidedItem) {
+            return navigation.push('SingleSelfGuided', {
+              itemId: selfGuidedItem.id,
+            });
+          }
+        }}
+        style={styles.container}>
+        <ImageBackground
+          source={{uri: selfGuidedItem.image}}
+          style={styles.backgroundImage}>
+          <View style={styles.infoContainer}>
+            <Text style={styles.title}>{selfGuidedItem.title}</Text>
+            <Text style={styles.location}>{selfGuidedItem.location}</Text>
+            <View style={styles.flexRow}>
+              <View style={styles.flexRow}>
+                <View style={styles.flexColumn}>
+                  <Text style={styles.subTitles}>Difficulty</Text>
+                  <Text style={styles.infobox}>
+                    {selfGuidedItem.difficulty}
+                  </Text>
+                </View>
+                <View style={styles.flexColumn}>
+                  <Text style={styles.subTitles}>Distance</Text>
+                  <Text style={styles.infobox}>{selfGuidedItem.distance}</Text>
+                </View>
+                <View style={styles.flexColumn}>
+                  <Text style={styles.subTitles}>Duration</Text>
+                  <Text style={styles.infobox}>
+                    {selfGuidedItem.duration} hours
+                  </Text>
+                </View>
+              </View>
+              <View style={styles.flexRow}>
+                <Image
+                  style={styles.contactSocialMediaIcon}
+                  source={require('../../assets/images/icFaveShareDefault.png')}
+                />
+                <TouchableOpacity
+                  onPress={async e => {
+                    try {
+                      if (
+                        savedIds.some(savedId => savedId === selfGuidedItem.id)
+                      ) {
+                        removeSaveId(selfGuidedItem.id);
+                      } else {
+                        addSaveId(selfGuidedItem.id);
+                      }
+                    } catch (error) {
+                      console.log(error);
```

Comment:
Please remove console log.

Reply:
Shouldn't I see the error in the console if I need to?

Reply:
Removed as requested.